c/detectAttlsPort.c

```diff
@@ -0,0 +1,138 @@
+    int part1 = (0xff000000 & (*inetAddress)->data.data4.addrBytes) >> 24;
+    printf("Resolved IP address as %d.%d.%d.%d\n", part1, part2, part3, part4);
+  }
+
+  /* TODO: No ipv6 resolution in getAddressByName yet, so nothing here either */
+  return TRUE;
+}
+
+static char *getKeywordArg(char *key, int argc, char **argv) {
+  for (int aa=1; aa<argc; aa++) {
+    if (!strcmp(argv[aa],key) &&
+        (aa+1 < argc)) {
+      // if --self-check then return the argument itself
+      if(!strcmp(argv[aa], "--self-check")) {
+        return argv[aa];
+      }
+      return argv[aa+1];
+    }
+  }
+  return NULL;
+}
+
+#define DETECT_ATTLS_PORT_STATUS_OK     0
+#define DETECT_ATTLS_PORT_STATUS_ERROR  8
+
+int main(int argc, char **argv) {
+  int status = DETECT_ATTLS_PORT_STATUS_OK;
+
+  if (argc == 1 || (argc == 2 && (strcmp(argv[1], "-h") == 0 || strcmp(argv[1], "--help") == 0))) {
+    printf("detect-attls-port - detects if AT-TLS is enabled in a live port.\n");
+    printf("  Format: [_BPX_JOBNAME=jobname] detect-attls-port --server hostname_or_ipv4 --port tcp_port --self-check [Use If you want to open a listening port,"
+           " and check if AT-TLS is enabled\n.");
+    printf("  Exit values: 0 AT-TLS has been enabled, 8 otherwise\n");
+    return DETECT_ATTLS_PORT_STATUS_OK;
+  }  
+
+  int returnCode = 0;
+  int reasonCode = 0;
+  int port = atoi(getKeywordArg("--port",argc,argv));
+  char *address = getKeywordArg("--host",argc,argv);
+  int selfCheck = FALSE;
+  if (getKeywordArg("--self-check",argc,argv)) {
+    selfCheck = TRUE;
+  }
+  
+  InetAddr *inetAddress = NULL;
+
+  if (!validateAddress(address, &inetAddress)) {
+    printf("Error: Invalid address given for --host\n");
+    return DETECT_ATTLS_PORT_STATUS_ERROR;
+  }
+  if (!port) {
+    printf("Error: No --port given\n");
+    return DETECT_ATTLS_PORT_STATUS_ERROR;
+  }
+
+  // self check if port has AT-TLS enabled as both server and client
+  if(selfCheck) {
+
+  } else { // Check destination port
+    int tlsFlags = 0;
+    Socket *serverSocket = tcpServer2(inetAddress, port, tlsFlags, &returnCode, &reasonCode);
+    
+    if (serverSocket) {
+      //printf("Bind succeeded (pointer=0x%p, rc=0x%x, rsn=0x%x)\n", serverSocket, returnCode, reasonCode);
+      int bpxrc=0, bpxrsn=0;
+      
+      struct TTLS_IOCTL ioc;              /* ioctl data structure          */
+      memset(&ioc,0,sizeof(ioc));         /* set all unused fields to zero */
+
+      arglen = sizeof(ioc);
+      sts = tcpIOControl(serverSocket, SIOCTTLSCTL, arglen, (char*)&ioc, &bpxrc, &bpxrsn);
+
+      int policy = ioc.TTLSi_Stat_Policy; // https://www.ibm.com/docs/en/zos/2.4.0?topic=ioctl-siocttlsctl-xc038d90b
+      switch (policy) {
+        case TTLS_POL_OFF:
+        case TTLS_POL_NO_POLICY:
+        case TTLS_POL_NOT_ENABLED:
+          printf("AT-TLS is not enabled\n");
+          break;
+        case TTLS_POL_ENABLED:
+        case TTLS_POL_APPLCNTRL:
+          printf("AT-TLS is enabled\n");
```

Comment:
The purpose of this program is to provide helpful advice on whether the enabled/disabled is normal, and if not, what the user should do about it.

So, let's change these printfs slightly.

Instead of just `ATTLS is enabled`
Try `No inbound AT-TLS rule identified for socket on ${ip}:${port} for user ${current_user} and job ${current job}`

You'll see elsehwere that we have some techniques (various) for detecting current user.
Current job, I'm not 100% sure on how to detect other than that the environment variable _BPX_JOBNAME is supposed to be one way to set it.